app/src/main/java/org/akvo/flow/activity/AppUpdateActivity.java

```diff
@@ -226,13 +255,10 @@ private File createFile(String location, String version) {
          * Downloads the apk file and stores it on the file system
          * After the download, a new notification will be displayed, requesting
          * the user to 'click to installAppUpdate'
-         *
-         * @param remoteFile
-         * @param surveyId
          */
         private boolean downloadApk(String location, String localPath) {
             Log.i(TAG, "App Update: Downloading new version " + mVersion + " from " + mUrl);
-            if (!StatusUtil.hasDataConnection(AppUpdateActivity.this)) {
+            if (activityWeakReference.get() != null && !StatusUtil.hasDataConnection(activityWeakReference.get())) {
```

Comment:
Same problem here - each call to get() could be the first to return a null, meaning that the Activity was just collected.